core/src/main/scripts/importer/validateData.py

```diff
@@ -2329,6 +2330,72 @@ def checkLine(self, data):
         else:
             self.fusion_entries[fusion_entry] = self.line_number
 
+
+class StructuralVariantValidator(Validator):
+
+    """Basic validation for fusion data. Validates:
+
+    1. Required column headers
+    2. Gene IDs
+    3. If there is a second gene, also validate Gene IDs of second gene
+    """
+
+    ### If the Entrez Gene id present, the Hugo Symbol can be skipped
+    REQUIRED_HEADERS = [
+        'Sample_ID',
+        'Annotation', # For example: EML4-ALK
+        'Event_Info', # For example: Fusion
+        ]
+    REQUIRE_COLUMN_ORDER = False
+
+    ALLOW_BLANKS = True
+
+    def __init__(self, *args, ** kwargs):
+        super(StructuralVariantValidator, self).__init__(*args, **kwargs)
+        self.structural_variant_entries = {}
+
+    def checkLine(self, data):
+
+        super(StructuralVariantValidator, self).checkLine(data)
+
+        def check_presence(column_name, self, data):
+            column_value = None
+            if column_name in self.cols:
+                column_value = data[self.cols.index(column_name)].strip()
+                # Treat the empty string or 'NA' as a missing value
+                if column_value in ['', 'NA']:
+                    column_value = None
+            return column_value
+
+        # Parse Hugo Symbol and Entrez Gene Id and compare them
+        hugo_symbol = check_presence('Site1_Hugo_Symbol', self, data)
+        entrez_gene_id = check_presence('Site1_Entrez_Gene_Id', self, data)
+        self.checkGeneIdentification(hugo_symbol, entrez_gene_id)
+
+        hugo_symbol = check_presence('Site2_Hugo_Symbol', self, data)
+        entrez_gene_id = check_presence('Site2_Entrez_Gene_Id', self, data)
+        self.checkGeneIdentification(hugo_symbol, entrez_gene_id)
+
+        # Validate if positions are given by chromosomal locations or exons
+        if data[self.cols.index('Event_Info')] == 'Fusion':
+            if set(['Site1_Chromosome', 'Site1_Position', 'Site2_Chromosome', 'Site2_Position']).issubset(set(self.cols)):
```

Comment:
should check the values instead of just checking whether the columns exist.

Reply:
Regarding ensembl transcript ids: if the validation is not added now, please add at least a WARNING telling that the system will assume canonical transcript id.

Reply:
:thinking: perhaps in any case we should also already retrieve the ensembl transcript and validate whether the given exon OR position falls within the given (or the assumed) transcript. You can use this web-service: https://rest.ensembl.org/lookup/id/ENST00000496384?content-type=application/json;expand=1   (although this ws will later be wrapped in a genomenexus call so the url will change)

Reply:
I implemented validation to check whether the exon is in the transcript.

Reply:
To do: validate if the transcript if valid

Reply:
Done. I also implemented validation for transcripts.